processor/src/test/java/org/mapstruct/ap/test/bugs/_2624/Issue2624Mapper.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2624;
+
+import java.util.Map;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public interface Issue2624Mapper {
+
+    Issue2624Mapper INSTANCE = Mappers.getMapper( Issue2624Mapper.class );
+
+    @Mapping( target = "department.id", source = "did")
```

Comment:
nice.. Just a thought.. you can't map nested sources, right? Since we don't know the type of the 'value' in the map if it's a `Map<String, Object>`.. 

I'm not sure how explicit the documentation is in this area (I remember that we discussed having a `sourceType` in `@Mapping`). And especially with reverting mappings (later on), this might to start to play a role. So also, perhaps we need to extend the documentation in this area.

I don't dare to ask 😄 .. but what happens if you specify a `@Mapping(target = ".", source = "did")`?

Reply:
>nice.. Just a thought.. you can't map nested sources, right? Since we don't know the type of the 'value' in the map if it's a `Map<String, Object>`..

No we can't, that has already been reported in #2553

>I don't dare to ask 😄 .. but what happens if you specify a `@Mapping(target = ".", source = "did")`?

You just did 😄 . I haven't tested this, but it will try to map whatever type the map value is to the current target. I don't really see a use case for this, so I'll rather wait for someone to complain about it 😄

Reply:
by all means 👍